### PR TITLE
Add some documentaiton about tests

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Documentation

### Description

Since we have only a little (sometimes misleading) information in `HACKING.md` we decide to create specific readme for the tests. We expect it will help to everyone understand how to easily write system tests. 

In case you will find any info not sufficient or missing, leave comment and I will update it.

### Checklist

- [x] Update documentation

### CODE REVIEW

**Comment by tomncooper** [comment]
Did you mean to include the `systemtest/pom.xml ` file?

**Comment by Frawless** [comment]
@tomncooper you mean include link for available profiles?

**Comment by tomncooper** [comment]
No I was just wondering why a docs PR required adding something to the systemtest pom file?

**Comment by Frawless** [comment]
Well, I added there a note, that `all` profile is used by default and for that we need this change. I can do it in separate PR but it will only add troubles with merging (another PR with minimal change).

**Comment by tomncooper** [comment]
Oh right. Make sense.

File: systemtest/TESTING.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+- [Use Remote Cluster](#use-remote-cluster)
+- [Helper Script](#helper-script)
+- [Running single test class](#running-single-test-class)
+- [Skip Teardown](#skip-teardown)
+
+<!-- /TOC -->
+
+## Pre-requisites 
+
+To run any system tests you need a Kubernetes or Openshift cluster available in your active kubernetes context. You can use [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/), [minishift](https://www.okd.io/minishift/), [oc cluster up](https://github.com/openshift/origin) or [CodeReady Contaners](https://github.com/code-ready/crc)
```
Comment (scholzj) [inline systemtest/TESTING.md]: Line per sentence will make it easier to review and comment not just now but also in the next PRs ;-)

File: HACKING.md [hunk 2]
```diff
@@ -234,75 +234,7 @@ The chart is also available in the release artifact as a tarball.
...
-
-#### Test Cluster
-
-The integration and system tests are run against a cluster specified in the environment variable `TEST_CLUSTER_CONTEXT`. If this variable is not set, kubernetes client will use currently active context. Otherwise will use context from kubeconfig with name specified by `TEST_CLUSTER_CONTEXT` variable.
-
-For example command `TEST_CLUSTER_CONTEXT=remote-cluster ./systemtest/scripts/run_tests.sh` will execute tests with cluster context `remote-cluster`. However, since system tests use command line `Executor` for some actions, make sure that you are using context from `TEST_CLUSTER_CONTEXT`.
-
-System tests uses admin user for some actions. You can specify admin user via variable `TEST_CLUSTER_ADMIN` (by default it use `developer` because `system:admin` cannot be used over remote connections).
+System tests has its own guide with more information. See [Testing Guide](https://github.com/strimzi/strimzi-kafka-operator/blob/master/systemtest/TESTING.md) document for more information.
```
Comment (scholzj) [inline HACKING.md]: Wouldn't it be better to have this in root and not under `systemtest`? I think it would be easier to find.

Comment (Frawless) [reply HACKING.md]: Make sense

Comment (scholzj) [reply HACKING.md]: I think you need to fix the link as well now ;-)

File: systemtest/TESTING.md (hunk 1 above)
Comment (Frawless) [reply systemtest/TESTING.md]: done

File: HACKING.md (hunk 2 above)
Comment (Frawless) [reply HACKING.md]: fixed

File: HACKING.md [hunk 3]
```diff
@@ -234,75 +234,7 @@ The chart is also available in the release artifact as a tarball.
...
-To set the log level of Strimzi for system tests need to add system property `STRIMZI_DEFAULT_LOG_LEVEL` with one of the following values: `ERROR`, `WARNING`, `INFO`, `DEBUG`, `TRACE`.
-
-#### Test Cluster
-
-The integration and system tests are run against a cluster specified in the environment variable `TEST_CLUSTER_CONTEXT`. If this variable is not set, kubernetes client will use currently active context. Otherwise will use context from kubeconfig with name specified by `TEST_CLUSTER_CONTEXT` variable.
-
-For example command `TEST_CLUSTER_CONTEXT=remote-cluster ./systemtest/scripts/run_tests.sh` will execute tests with cluster context `remote-cluster`. However, since system tests use command line `Executor` for some actions, make sure that you are using context from `TEST_CLUSTER_CONTEXT`.
-
-System tests uses admin user for some actions. You can specify admin user via variable `TEST_CLUSTER_ADMIN` (by default it use `developer` because `system:admin` cannot be used over remote connections).
+System tests has its own guide with more information. See [Testing Guide](https://github.com/strimzi/strimzi-kafka-operator/blob/master/TESTING.md) document for more information.
```
Comment (scholzj) [inline HACKING.md]: Actually ... could we make this link relative? To work in branches / tags?

File: TESTING.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+## Package Structure 
+
+Systemtest package is divided into `main` and `test` as usually. 
+In `main` you can find all support classes, which are used in the tests. 
+Modules worth for mention are:
+
+* **annotations** — we have our own `@OpenShiftOnly` annotation, which check if current cluster is Openshift or not. All other annotations should be stored here.
+* **clients** — clients implementations used in tests.
+* **matchers** — it contains our matcher implementation for checking cluster operator logs. For more info see [matcher details](#matcher).
+* **utils** — a lot of actions are the same for most of the tests, and we share them through utils class ad static methods. You can find here most of the useful methods.
```
Comment (granzoto) [inline TESTING.md]: typo :  class **ad** static methods

**Pull request merged**